compiler/ccgtypes.nim

```diff
@@ -1350,17 +1350,19 @@ proc genDisplay(t: PType, depth: int): Rope =
 
 proc genTypeInfoV2Impl(m: BModule; t, origType: PType, name: Rope; info: TLineInfo) =
   cgsym(m, "TNimTypeV2")
-  m.s[cfsStrData].addf("N_LIB_PRIVATE TNimTypeV2 $1;$n", [name])
+  if m.compileToCpp: # todo how to handle C++?
```

Comment:
C++ doesn't support forward variables without external linkage, which is not desirable. Probably we should do global assignment for C++.

Reply:
How about keep original code for C++, I can improve C++ codegen in the following pRs.

Reply:
Sounds good.